### PR TITLE
Refactor and simplify python builds

### PR DESCRIPTION
**Pull request by jwhonce**
* podman RPM will install podman and pypodman eggs
* pypodman namespaced in site-packages
* version numbers pulled from requirements.txt
* Makefiles updated to run all operations with PYTHON when set
* Fixed bugs with supporting python3.4

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by jwhonce** [comment]
Future PR will support pylint checking against different versions of Python

@lsm5 Please review spec file.  I'll remove WIP when you and I agree on the installs. 

Note: ```make install.python`` is intended to install on the system not build rpms etc.

**Comment by rhatdan** [comment]
I have not looked at the code, but if the podman package installs the python code, doesn't this mean podman will require python?  Definitely not something we want.

File: contrib/spec/podman.spec.in [hunk 1]
```diff
@@ -502,6 +474,7 @@ export GOPATH=%{buildroot}/%{gopath}:$(pwd)/vendor:%{gopath}
 %license LICENSE
 %doc README.md CONTRIBUTING.md install.md code-of-conduct.md transfer.md
 %{_bindir}/%{name}
+%{_bindir}/py%{name}
```
Comment (lsm5) [inline contrib/spec/podman.spec.in]: sorry about suggesting this earlier. This line should actually be in the python-podman or python-pypodman (if we're doing separate rpms for these)

Comment (rhatdan) [reply contrib/spec/podman.spec.in]: Yes lets keep pypodman in the python3-podman package.

**Comment by lsm5** [comment]
@jwhonce btw, doesn't look like the spec successfully builds. With your spec (and also with the fedora dist-git spec), I see: 

-----------------------------------------------------
/usr/bin/make -C contrib/python/podman install
make[1]: Entering directory '/home/lsm5/repositories/libpod/contrib/spec/BUILD/libpod-341364dc5bf716e328295659bb9e541bcf877297/contrib/python/podman'
/usr/bin/python3 setup.py install
running install
error: can't create or remove files in install directory

The following error occurred while trying to add or remove files in the
installation directory:

    [Errno 13] Permission denied: '/usr/lib/python3.6/site-packages/test-easy-install-30254.write-test'

The installation directory you specified (via --install-dir, --prefix, or
the distutils default setting) was:

    /usr/lib/python3.6/site-packages/

Perhaps your account does not have write access to this directory?  If the
installation directory is a system-owned directory, you may need to sign in
as the administrator or "root" account.  If you do not have administrative
access to this machine, you may wish to choose a different installation
directory, preferably one that is listed in your PYTHONPATH environment
variable.

For information on other options, you may wish to consult the
documentation at:

  https://setuptools.readthedocs.io/en/latest/easy_install.html

Please make the appropriate changes for your system and try again.

make[1]: *** [Makefile:17: install] Error 1
make[1]: Leaving directory '/home/lsm5/repositories/libpod/contrib/spec/BUILD/libpod-341364dc5bf716e328295659bb9e541bcf877297/contrib/python/podman'
make: *** [Makefile:245: install.python] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.yEkgAp (%install)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.yEkgAp (%install)
-----------------------------------------------------

**Comment by jwhonce** [comment]
@lsm5 the site-package directory is selected based on the version of the python command you are running, so you should be running 3.6 given that output.  Currently the Makefiles assume you are root to do an install.  

Seeing @rhatdan 's comment, I'll add another commit to this break out a python-podman rpm from the podman rpm.  /cc @baude

**Comment by lsm5** [comment]
> the site-package directory is selected based on the version of the python command you are running, so you should be running 3.6 given that output. Currently the Makefiles assume you are root to do an install.

So, on fedora 28, there's just 1 python3 and that's python 3.6. The rpm spec upstream uses `python3 setup.py install --root %{buildroot}` . Fedora's dist-git also uses `%py3_install` which is pretty much the same as setup.py install --root %{buildroot}, so neither of those should need root. Anyway, let me look further and get back.

**Comment by jwhonce** [comment]
@lsm5 Your error report starts with "/usr/bin/make -C contrib/python/podman install"

which on this PR installs in the system libraries not BUILDROOT so would require root.  On your examples what is BUILDROOT 's value?

**Comment by lsm5** [comment]
@jwhonce The error I posted above was with your spec file in contrib/spec/ . It uses `--root %{buildroot}`. Are you able to get a successful `rpmbuild -ba podman.spec` using that spec file and with a non-root user?

**Comment by lsm5** [comment]
fwiw, `rpm --eval %{buildroot}` = /home/lsm5/repositories/daily-rebuild/BUILDROOT/%{NAME}-%{VERSION}-%{RELEASE}.x86_64 on my machine.

**Comment by jwhonce** [comment]
@rhatdan This PR is step one for removing python as a dependency of podman.

**Comment by jwhonce** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
@jwhonce I was thinking of this being only a specfile issue, not requiring code changes.

**Comment by rhatdan** [comment]
LGTM
@lsm5 @mheon @baude PTAL

**Comment by mheon** [comment]
LGTM

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

File: Makefile [hunk 2]
```diff
@@ -206,7 +205,7 @@ changelog:
 	$(shell cat $(TMPFILE) >> changelog.txt)
 	$(shell rm $(TMPFILE))
 
-install: .gopathok install.bin install.man install.cni install.systemd
+install: .gopathok install.bin install.man install.cni install.systemd install.python
```
Comment (lsm5) [inline Makefile]: umm, can we remove install.python from here? Given that we intend to separate out the python components long term? Also, can we include install.completions here instead?

File: contrib/python/podman/Makefile [hunk 3]
```diff
@@ -2,20 +2,30 @@ PYTHON ?= /usr/bin/python3
 
 .PHONY: python-podman
 python-podman:
-	$(PYTHON) setup.py bdist
+	$(PYTHON) setup.py sdist bdist
+
+.PHONY: lint
+lint:
+	$(PYTHON) -m pylint podman
```
Comment (cevich) [inline contrib/python/podman/Makefile]: Just a heads-up:  @edsantiago and I have had almost nothing but headaches with pylint.  It "works", just a PITA to maintain for CI:  There are sometimes incompatible variations in message-codes/behaviors across different distros (ubuntu vs RHEL vs Fedora).  Might be worth looking for a different tool in the near-term.

**Comment by cevich** [comment]
LGTM...except for ``Makefile``/``setup.py`` black-magic I encountered, couldn't review that out of fear for my firstborn :grin:

File: contrib/python/podman/README.md [hunk 4]
```diff
@@ -6,11 +6,12 @@ See [libpod](https://github.com/projectatomic/libpod)
...
 
-To build the podman egg:
+To build the podman egg and install as user:
 
 ```sh
-cd ~/libpod/contrib/python
-python3 setup.py clean -a && python3 setup.py bdist
+cd ~/libpod/contrib/python/podman
+python3 setup.py clean -a && python3 setup.py sdist bdist
+python3 setup.py install --user
```
[the remark this answers is not in the document]
Comment (jwhonce) [reply contrib/python/podman/README.md]: @cevich Notice the --user option?  The makefile installs into the system areas.  Seemed a good way to segregate them.

File: contrib/python/podman/Makefile (hunk 3 above)
Comment (jwhonce) [reply contrib/python/podman/Makefile]: @cevich @edsantiago Agreed pylint is very picky.  None of the other "linters" actually lock down the exact version of python as well.  kwargs to methods being the big win with pylint. As a compromise how about a pylint target and have the lint target use yapf or pyflakes?  Something like clobber vs. clean.

**Comment by jwhonce** [comment]
@lsm5 rhel8 should work as well.  To run the code you need to install varlink via pip3 as there is no rpm yet.

**Pull request closed without merging**